app/src/main/java/com/mybus/model/BusRoute.java

```diff
@@ -53,6 +56,60 @@ public BusRoute(Integer mIdBusLine, String mBusLineName, Integer mBusLineDirecti
         this.mDestinationBusStopDistanceToDestination = mDestinationBusStopDistanceToDestination;
     }
 
+    protected BusRoute(Parcel in) {
+        mBusLineName = in.readString();
+        mBusLineColor = in.readString();
+        mStartBusStopLat = in.readString();
+        mStartBusStopLng = in.readString();
+        mStartBusStopStreetName = in.readString();
+        mDestinationBusStopLat = in.readString();
+        mDestinationBusStopLng = in.readString();
+        mDestinationBusStopStreetName = in.readString();
+        mDestinationBusStopStreetNumber = in.readInt();
+        mStartBusStopStreetNumber = in.readInt();
+        mDestinationBusStopDistanceToDestination = in.readDouble();
+        mStartBusStopDistanceToOrigin = in.readDouble();
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        dest.writeString(mBusLineName);
+        dest.writeString(mBusLineColor);
+        dest.writeString(mStartBusStopLat);
+        dest.writeString(mStartBusStopLng);
+        dest.writeString(mStartBusStopStreetName);
+        dest.writeString(mDestinationBusStopLat);
+        dest.writeString(mDestinationBusStopLng);
+        dest.writeString(mDestinationBusStopStreetName);
+        dest.writeInt(mDestinationBusStopStreetNumber);
+        dest.writeInt(mStartBusStopStreetNumber);
+        if (mDestinationBusStopDistanceToDestination == null) {
+            mDestinationBusStopDistanceToDestination = -1d;
```

Comment:
could be another fix: http://stackoverflow.com/questions/5905105/how-to-serialize-null-value-when-using-parcelable-interface